lib/arqo/query.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module ARQO
+  # Parent class for query objects
+  class Query
+    attr_reader :relation
+    delegate_missing_to :relation
+
+    def initialize(relation = derived_relation)
+      @relation = relation
+      @relation = @relation.extending(scope_module) if Object.const_defined?(scope_module_name)
```

Comment:
or maybe not even making it optional so we force people to define `Scope` ? 🤔 at this point I don't see any benefit to not writing the method directly under the `Scope` module..

Reply:
That's right, I'd rather have end users define the query methods directly in the Scope module, always. 

This way we also make it more straightforward, because there would be only one way of defining query methods. Least surprise (?)